NoName-ForTeam/photo-fiesta
# Feature/proj 145

# Profile API Integration and Data Management

## Changes Made

### 1. Profile API and Types Update
- Added constants for profile API endpoints in `config/api.ts`
- Updated `baseApi.ts` to include new API endpoints for profile operations
- Enhanced `profile.types.ts`:
  - Added JSDoc comments for better code documentation
  - Updated types and introduced interfaces for improved type safety
  - Restructured types to better represent the API response and request structures

### 2. User Data and Avatar Upload Implementation
- Implemented functionality to upload user profile data to the server
- Added avatar upload capability, including handling of both JPEG and PNG formats
- Created `imageUtils.ts` for converting image Data URLs to a format suitable for server upload

### 3. Code Refactoring and Optimization
- Refactored `useGeneralInfo` hook to use the new API endpoints and types
- Optimized image processing logic for better performance and reusability

## Detailed Changes

### Profile API Integration
- Added new endpoints in `baseApi.ts` for operations like:
  - Fetching user profile
  - Updating user profile
  - Uploading avatar
  - Deleting avatar and profile

### Types and Interfaces Update
- Updated `ProfileResponse` and `ProfileSettings` interfaces in `profile.types.ts`
- Added comprehensive JSDoc comments to explain the purpose and structure of each type/interface

### Image Upload Utility
- Created `imageUtils.ts` with a `prepareImageForUpload` function to handle image data conversion
- This utility supports both JPEG and PNG formats and prepares the image data for server upload

### Profile Update Logic
- Enhanced `useGeneralInfo` hook to handle both profile data and avatar updates
- Implemented error handling and success notifications for profile update operations

## Testing Instructions
1. Test profile data fetching and display
2. Attempt to update profile information and verify the changes are reflected
3. Upload different image formats (JPEG, PNG) as avatars and ensure they are processed correctly
4. Verify error handling by intentionally causing errors (e.g., network disconnection during upload)

## Notes
- Ensure the backend API endpoints match the new constants defined in the frontend
- The image upload utility may need adjustments based on specific backend requirements

Please review and test thoroughly, focusing on the integration points between the new API calls and the existing UI components.
